### PR TITLE
Feat: Storybook Updates

### PR DESCRIPTION
**Pull request by renatamottam**
## What's the purpose of this pull request?
This PR updates the following Storybook docs to follow our current `mdx` structure:
- Link
- Select
- ProductCard
- Badge

## How to test it?
Check all the docs page on Storybook.

## Checklist
**PR Title and Commit Messages**
- [x] PR title and commit messages follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specification
  - Available prefixes: `feat`, `fix`, `chore`, `docs`, `style`, `refactor`, `perf`, and `test`

**PR Description**
- [x] Added a label according to the PR goal - `Breaking change`, `Features`, `Bug fixes`, `Chore`, `Documentation`, `Style changes`, `Refactoring`, `Performance`, and `Test`
- [x] Added the component, hook, or path name in-between backticks (\`\`) - *if applicable, e.g., `ComponentName` component, `useWindowDimensions` hook*

**Documentation**
- [x] PR description
- [x] Added to/Updated the Storybook - *if applicable*

### CODE REVIEW

**Review by lucasfp13 — APPROVED** [review]
Nice 💯 

I left just a couple of suggestions.

File: src/components/product/ProductCard/ProductCard.tsx [hunk 1]
```diff
@@ -22,9 +22,21 @@ type Variant = 'wide' | 'default'
 export interface ProductCardProps {
   product: ProductSummary_ProductFragment
   index: number
+  /**
+   * Set a border to the component.
```
Comment (hellofanny) [inline src/components/product/ProductCard/ProductCard.tsx:26]:
```suggestion
   * Sets a border to the component.
```

File: src/components/product/ProductCard/ProductCard.tsx [hunk 2]
```diff
@@ -22,9 +22,21 @@ type Variant = 'wide' | 'default'
...
+  /**
+   * Controls the component's size.
+   */
   variant?: Variant
+  /**
+   * Specifies the ProductCard image's aspect ratio.
+   */
   aspectRatio?: number
+  /**
+   * Enable a ButtonBuy to the component.
```
Comment (hellofanny) [inline src/components/product/ProductCard/ProductCard.tsx:38]:
```suggestion
   * Enables a ButtonBuy to the component.
```

File: src/components/product/ProductCard/ProductCard.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,412 @@
...
+      aspectRatio: 1.5,
+      ButtonBuy: false,
+    }}
+  >
+    {TemplateWide.bind({})}
+  </Story>
+</Canvas>
+
+<TokenTable>
+  <TokenRow token="--fs-product-card-wide-padding" value="0" />
```
Comment (hellofanny) [inline src/components/product/ProductCard/ProductCard.stories.mdx:401]: I don't think we've set this value in [product-card.module.scss](https://github.com/vtex-sites/nextjs.store/pull/271/files#diff-4a4720a8e1418f3f548a29ab35618828c79566b72509303f656bcf1fbd2233a0)

File: src/components/product/ProductCard/product-card.module.scss [hunk 4]
```diff
@@ -34,24 +34,30 @@
...
+  --fs-product-card-description-size         : var(--fs-text-size-legend);
 
   // Subtitle
-  --fs-product-card-subtitle-color      : var(--fs-color-text-light);
-  --fs-product-card-subtitle-size       : var(--fs-text-size-tiny);
+  --fs-product-card-subtitle-color           : var(--fs-color-text-light);
+  --fs-product-card-subtitle-size            : var(--fs-text-size-tiny);
+
+  // Wide
+  --fs-product-card-wide-padding             : var(--fs-product-card-wide-padding);
```
Comment (hellofanny) [inline src/components/product/ProductCard/product-card.module.scss:57]: Can you double check this variable value? `--fs-product-card-wide-padding`

I don't think we have set a value for it here. 🤔 
But we are listing in the [doc](https://github.com/vtex-sites/nextjs.store/pull/271/files#r996617032)

File: src/components/ui/Badge/DiscountBadge.tsx [hunk 5]
```diff
@@ -5,12 +5,25 @@ import { useDiscountPercent } from 'src/sdk/product/useDiscountPercent'
...
+  /**
+   * Specifies old price before discount.
+   */
   listPrice: number
+  /**
+   * Specifies current price with discount applied.
+   */
   spotPrice: number
+  /**
+   * Specifies the component size as big.
```
Comment (hellofanny) [inline src/components/ui/Badge/DiscountBadge.tsx:17]:
```suggestion
   * Sets the component size as big.
```

File: src/components/ui/Badge/DiscountBadge.tsx [hunk 6]
```diff
@@ -5,12 +5,25 @@ import { useDiscountPercent } from 'src/sdk/product/useDiscountPercent'
...
+   */
   big?: boolean
-  // Set limit percentage value to consider a low discount.
+  /**
+   * Set limit percentage value to consider a low discount.
```
Comment (hellofanny) [inline src/components/ui/Badge/DiscountBadge.tsx:21]:
```suggestion
   * Sets the limit percentage value to consider a low discount.
```

File: src/components/ui/Badge/DiscountBadge.tsx [hunk 7]
```diff
@@ -5,12 +5,25 @@ import { useDiscountPercent } from 'src/sdk/product/useDiscountPercent'
...
+   */
   thresholdLow?: number
-  // Set limit percentage value to consider a high discount
+  /**
+   * Set limit percentage value to consider a high discount.
```
Comment (hellofanny) [inline src/components/ui/Badge/DiscountBadge.tsx:25]:
```suggestion
   * Sets the limit percentage value to consider a high discount.
```

File: src/components/product/ProductCard/ProductCard.stories.mdx (hunk 3 above)
Comment (renatamottam) [reply src/components/product/ProductCard/ProductCard.stories.mdx:401]: we use it to overwrite some tokens used as structural styles for `ProductCards`
image

File: src/components/ui/Badge/Badge.stories.mdx [hunk 8]
```diff
@@ -0,0 +1,410 @@
...
+import {
+  TokenTable,
+  TokenRow,
+  TokenDivider,
+  SectionList,
+  SectionItem,
+} from 'src/../.storybook/components'
+
+<Meta
+  title="Molecules/Badge/Default"
```
[the remark this answers is not in the document]
Comment (renatamottam) [reply src/components/ui/Badge/Badge.stories.mdx:15]: In this scenario, this is the regular badge doc, not an overview of all badges, so I didn't know what would be the best name for the page.

File: src/components/product/ProductCard/ProductCard.tsx [hunk 9]
```diff
@@ -22,9 +22,21 @@ type Variant = 'wide' | 'default'
 export interface ProductCardProps {
   product: ProductSummary_ProductFragment
   index: number
+  /**
+   * Sets a border to the component.
+   */
   bordered?: boolean
+  /**
+   * Controls the component's size.
```
Comment (hellofanny) [inline src/components/product/ProductCard/ProductCard.tsx:30]:
```suggestion
   * Sets the component's size.
```

File: src/components/ui/Badge/Badge.tsx [hunk 10]
```diff
@@ -18,7 +24,7 @@ type ActionableBadge =
       onClose?: never
     }
 
-type Props = {
+export type BadgeProps = {
   big?: boolean
```
Comment (hellofanny) [inline src/components/ui/Badge/Badge.tsx:28]: Should we add descriptions for those props?

Comment (hellofanny) [inline src/components/ui/Badge/Badge.tsx:28]:
```suggestion
  /**
   * Sets the component's size.
   */
  big?: boolean
```

File: src/components/ui/Badge/DiscountBadge.stories.mdx [hunk 11]
```diff
@@ -0,0 +1,171 @@
...
+
+<Meta title="Molecules/Badge/DiscountBadge" component={DiscountBadge} />
+
+export const Template = (args) => <DiscountBadge {...args} />
+
+<header>
+
+# Discount Badge
+
+A `Badge` used to show product discounts. You can see its usage on [ProductCard](?path=/docs/molecules-productcard--overview-default).
```
Comment (hellofanny) [inline src/components/ui/Badge/DiscountBadge.stories.mdx:19]:
```suggestion
A custom `Badge` that display product's discounts. You can see its usage on [ProductCard](?path=/docs/molecules-productcard--overview-default).
```

File: src/components/ui/Badge/DiscountBadge.stories.mdx [hunk 12]
```diff
@@ -0,0 +1,171 @@
...
+    args={{
+      big: false,
+      listPrice: 45,
+      spotPrice: 40,
+    }}
+  >
+    {Template.bind({})}
+  </Story>
+</Canvas>
+
```
Comment (hellofanny) [inline src/components/ui/Badge/DiscountBadge.stories.mdx:39]: Maybe we can also add the other variants here (medium and high discount) 

We've been listing the other variants in the overview section whenever possible. WDYT?

image

**Pull request merged**